### PR TITLE
Versioned configs

### PR DESCRIPTION
**Pull request by yportnoy**

### CODE REVIEW

**Comment by jgillis01** [comment]
@yportnoy Bump Version?

File: spec/unit/config_spec.rb [hunk 1]
```diff
@@ -26,7 +27,8 @@
         :max_workers_per_host => nil,
         :cluster_name => "unit-test-cluster",
         :environment_name => "unit-test",
-        :manage_worker_heartbeats => true
+        :manage_worker_heartbeats => true,
+        :version_hash => `git rev-parse --verify HEAD`.strip
```
Comment (jgillis01) [inline spec/unit/config_spec.rb]: Possibly use the git gem to get this information?

File: lib/resque/cluster/config.rb [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        end
+      end
+
+      private
+
+      def config_version(global_config_path)
+        return unless verified?
+        directory_name = File.dirname(global_config_path)
+        if Dir.exists?(directory_name)
+          output = `cd #{directory_name}; git rev-parse --verify HEAD; echo $?`.split
```
Comment (jgillis01) [inline lib/resque/cluster/config.rb]: @yportnoy Possibly use git gem for this?

Comment (yportnoy) [inline lib/resque/cluster/config.rb]: Tried the gem, fails to recursively find the .ssh folder if not in the root

**Pull request merged**